python/py_bindings.h

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2025 Spglib team
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <optional>
+
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+namespace py = pybind11;
+namespace spglib {
+using array_double =
+    py::array_t<double, py::array::c_style | py::array::forcecast>;
+using array_int = py::array_t<int, py::array::c_style | py::array::forcecast>;
+using array_uintp =
+    py::array_t<uintptr_t, py::array::c_style | py::array::forcecast>;
+using array_size_t =
+    py::array_t<size_t, py::array::c_style | py::array::forcecast>;
+py::tuple version_tuple();
+py::str version_string();
+py::str version_full();
+py::str commit();
+std::optional<py::list> dataset(array_double lattice, array_double positions,
+                                array_int atom_types, py::int_ hall_number,
+                                py::float_ symprec, py::float_ angle_tolerance);
+std::optional<py::list> layerdataset(array_double lattice,
```

Comment:
Can you change this into `layer_dataset` to be consistent with `magnetic_dataset`?

Reply:
Right now I have them mirroring the exposed python interface. It is a private module, so I think we can adjust both of them for that